server/src/static/assets/js/filter.js

```diff
@@ -13,3 +13,98 @@ document.addEventListener('DOMContentLoaded', function() {
         });
     });
 });
+
+function sortTable(header, table) {
+  var SORTABLE_STATES = {
+    none: 0,
+    ascending: -1,
+    descending: 1,
+    ORDER: ['none', 'ascending', 'descending'],
+  };
+
+  // Get index of column based on position of header cell in <thead>
+  // We assume there is only one row in the table head.
+  var col = [].slice.call(table.tHead.rows[0].cells).indexOf(header);
+
+  // Based on the current aria-sort value, get the next state.
+  var newOrder = SORTABLE_STATES.ORDER.indexOf(header.getAttribute('aria-sort')) + 1;
+  newOrder = newOrder > SORTABLE_STATES.ORDER.length - 1 ? 0 : newOrder;
+  newOrder = SORTABLE_STATES.ORDER[newOrder];
+
+  // Reset all header sorts.
+  var headerSorts = table.querySelectorAll('[aria-sort]');
+
+  for (var i = 0, ii = headerSorts.length; i < ii; i += 1) {
+    headerSorts[i].setAttribute('aria-sort', 'none');
+  }
+
+  // Set the new header sort.
+  header.setAttribute('aria-sort', newOrder);
+
+  // Get the direction of the sort and assume only one tbody.
+  // For this example only assume one tbody.
+  var direction = SORTABLE_STATES[newOrder];
+  var body = table.tBodies[0];
+
+  // Convert the HTML element list to an array.
+  var newRows = [].slice.call(body.rows, 0);
+
+  // If the direction is 0 - aria-sort="none".
+  if (direction === 0) {
+    // Reset to the default order.
+    newRows.sort(function (a, b) {
+      return a.getAttribute('data-index') - b.getAttribute('data-index');
+    });
+  } else {
+    // Sort based on a cell contents
+    newRows.sort(function (rowA, rowB) {
+      // Trim the cell contents.
+      var contentA = rowA.cells[col].textContent.trim();
+      var contentB = rowB.cells[col].textContent.trim();
+
+      // Based on the direction, do the sort.
+      //
+      // This example only sorts based on alphabetical order, to sort based on
+      // number value a more specific implementation would be needed, to provide
+      // number parsing and comparison function between text strings and numbers.
+      return contentA < contentB ? direction : -direction;
+    });
+  }
+  // Append each row into the table, replacing the current elements.
+  for (i = 0, ii = body.rows.length; i < ii; i += 1) {
+    body.appendChild(newRows[i]);
+  }
```

Comment:
@plars I'm a little worried about this one. Frankly, I'm not sure what would happen here. Just worried that every time we sort, the number of rows doubles. Unless they are replaced somehow and I missed it

Reply:
@omar-selo From trying it locally, I can confirm that it doesn't double the table length each time. According to https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild - it looks like if it refers to an existing node, it moves it rather than creating a new one:
"Note: If the given child is a reference to an existing node in the document, appendChild() moves it from its current position to the new position."